### PR TITLE
refactor: apply qa findings for performance and architecture

### PR DESCRIPTION
**Pull request by BjornMelin**
## Summary
- Relocated database logic from scraper orchestrator to appropriate service layer for better separation of concerns
- Optimized UI tab filtering to use efficient database queries instead of in-memory filtering
- Centralized application status constants to follow DRY principle and eliminate code duplication

### Architecture Improvements
- **Moved `bulk_get_or_create_companies`** from `src/scraper.py` to `src/services/company_service.py` as a static method
- **Updated scraper imports** to use the relocated service method
- **Improved separation of concerns** by keeping database operations in the service layer

### Performance Optimizations
- **Database-level filtering** for Favorites and Applied tabs instead of client-side filtering
- **Added helper functions** `_get_favorites_jobs()` and `_get_applied_jobs()` for targeted queries
- **Reduced memory overhead** by querying only needed records per tab

### Code Quality & Maintainability
- **Centralized `APPLICATION_STATUSES`** constant in `src/constants.py`
- **Updated job card component** to use the centralized constant
- **Eliminated hardcoded status lists** throughout the UI layer

## Testing
- **40+ comprehensive pytest tests** covering all refactored functionality
- **Integration tests** validating end-to-end workflows
- **Performance tests** with large datasets (50+ companies/jobs)
- **Edge case coverage** including race conditions and error handling

## Technical Details
- All changes maintain backward compatibility and existing functionality
- Database query optimization improves scalability with large datasets
- Code follows project standards with full ruff linting compliance
- Test coverage includes realistic scenarios mirroring production usage

## Verification
- ✅ All existing tests pass
- ✅ New comprehensive test suite (40+ tests) passes
- ✅ Ruff linting and formatting applied
- ✅ Functionality verified identical to previous implementation
- ✅ Performance improvements validated

Ready for production deployment.

## Summary by Sourcery

Apply QA-driven architecture and performance improvements by relocating database logic to service layer, streamlining UI filtering through database queries, consolidating status constants, and bolstering test coverage

Enhancements:
- Move bulk_get_or_create_companies into CompanyService with bulk operation optimizations and race condition handling
- Refactor scraper to invoke CompanyService.bulk_get_or_create_companies for database operations
- Optimize UI tabs to query favorites and applied jobs directly via JobService instead of in-memory filtering
- Centralize application status values in APPLICATION_STATUSES constant and update UI components to use it

Tests:
- Add extensive unit tests for JobService filters, CompanyService bulk operations, and application status constants validation
- Introduce integration tests covering end-to-end workflows, UI filtering, constants access, and performance scenarios

### CODE REVIEW

File: src/services/company_service.py [hunk 1]
```diff
@@ -611,3 +615,91 @@ def get_active_companies_count() -> int:
         except Exception:
             logger.exception("Failed to get active companies count")
             raise
+
+    @staticmethod
+    @performance_monitor
+    def bulk_get_or_create_companies(
```
[the remark this answers is not in the document]
Comment (BjornMelin) [reply src/services/company_service.py:621]: Skipped because it would break SQLite compatibility and violate KISS principle
